packages/styletron-react/src/index.js

```diff
@@ -559,7 +557,10 @@ export function resolveStyle(
   let result = getInitialStyle();
   let i = reducers.length;
   while (i--) {
-    result = reducers[i].reducer(result, props);
+    const reducer = reducers[i];
+    result = reducer.assignmentCommutative
```

Comment:
This will incur some (potentially insignificant) runtime penalty. Might worth some casting to avoid it.

Reply:
we could add a default value to `props` of an empty object. this way, `reducer.reducer` can always be called with both arguments; removing the need for a runtime check

Reply:
I prefer to typecast since we're only satisfying flow in this case, is isolated to the utility function, and we know that calling with additional arguments is a safe action. The main issue is if the two reducer types significantly diverge. I tested locally with the def below passing, and even changed the reducer type if the container types to see if an error would be surfaced (which it did). What's your preference? 

```js
const reducer = (reducers[i].reducer: (StyleObject, Object) => StyleObject);
result = reducer(result, props);
```

Reply:
Thanks, I like your suggestion. Just commited that